api/kv_v2.go

```diff
@@ -38,17 +38,40 @@ type KVVersionMetadata struct {
 	Destroyed    bool      `mapstructure:"destroyed"`
 }
 
-// Currently supported options: WithCheckAndSet
+// Currently supported options: WithOption, WithCheckAndSet, WithMethod
 type KVOption func() (key string, value interface{})
 
+const (
+	KVOptionCheckAndSet = "cas"
+	KVOptionMethod      = "method"
+)
+
+// WithOption can optionally be passed to provide generic options for a
+// KV request. Valid keys and values depend on the type of request.
+func WithOption(key string, value interface{}) KVOption {
+	return func() (string, interface{}) {
+		return key, value
+	}
+}
+
 // WithCheckAndSet can optionally be passed to perform a check-and-set
-// operation. If not set, the write will be allowed. If cas is set to 0, a
-// write will only be allowed if the key doesn't exist. If set to non-zero,
-// the write will only be allowed if the key’s current version matches the
-// version specified in the cas parameter.
+// operation on a KV request. If not set, the write will be allowed.
+// If cas is set to 0, a write will only be allowed if the key doesn't exist.
+// If set to non-zero, the write will only be allowed if the key’s current
+// version matches the version specified in the cas parameter.
 func WithCheckAndSet(cas int) KVOption {
 	return func() (string, interface{}) {
-		return "cas", cas
+		return KVOptionCheckAndSet, cas
+	}
+}
+
+// WithMethod can optionally be passed to dictate which type of
+// patch to perform in a Patch request. If set to "patch", then an HTTP PATCH
+// request will be issued. If set to "rw", then a read will be performed,
+// then a local update, followed by a remote update. Defaults to "patch".
+func WithMethod(method string) KVOption {
+	return func() (string, interface{}) {
+		return KVOptionMethod, method
 	}
 }
```

Comment:
Might be more clear to wrap your new WithOption function:

```suggestion
func WithMethod(method string) KVOption {
	return WithOption(KVOptionMethod, method)
}
```